core/src/main/java/com/orientechnologies/orient/core/storage/cache/local/OWOWCache.java

```diff
@@ -1280,97 +1378,96 @@ private void removeCachedPages(int fileId) {
+      throws IOException {
     final long fileId = composeFileId(id, intId);
-    final OClosableEntry<Long, OFileClassic> entry = files.acquire(fileId);
+
     try {
-      final OFileClassic fileClassic = entry.get();
-      if (fileClassic == null)
-        throw new IllegalArgumentException("File with id " + intId + " not found in WOW Cache");
+      final OClosableEntry<Long, OFileClassic> entry = files.acquire(fileId);
+      try {
+        final OFileClassic fileClassic = entry.get();
+        if (fileClassic == null)
+          throw new IllegalArgumentException("File with id " + intId + " not found in WOW Cache");
 
-      final OLogSequenceNumber lastLsn;
-      if (writeAheadLog != null)
-        lastLsn = writeAheadLog.getFlushedLsn();
-      else
-        lastLsn = new OLogSequenceNumber(-1, -1);
+        final long firstPageStartPosition = startPageIndex * pageSize;
+        final long firstPageEndPosition = firstPageStartPosition + pageSize;
 
-      final long firstPageStartPosition = startPageIndex * pageSize;
-      final long firstPageEndPosition = firstPageStartPosition + pageSize;
+        if (fileClassic.getFileSize() >= firstPageEndPosition) {
+          final OSessionStoragePerformanceStatistic sessionStoragePerformanceStatistic = performanceStatisticManager
+              .getSessionPerformanceStatistic();
+          if (sessionStoragePerformanceStatistic != null) {
+            sessionStoragePerformanceStatistic.startPageReadFromFileTimer();
+          }
 
-      if (fileClassic.getFileSize() >= firstPageEndPosition) {
-        final OSessionStoragePerformanceStatistic sessionStoragePerformanceStatistic = performanceStatisticManager
-            .getSessionPerformanceStatistic();
-        if (sessionStoragePerformanceStatistic != null) {
-          sessionStoragePerformanceStatistic.startPageReadFromFileTimer();
-        }
+          int pagesRead = 0;
 
-        int pagesRead = 0;
+          final OLogSequenceNumber lastLsn = writeAheadLog == null ? new OLogSequenceNumber(-1, -1) : writeAheadLog.getFlushedLsn();
 
-        try {
-          if (pageCount == 1) {
-            final ByteBuffer buffer = bufferPool.acquireDirect(false);
-            assert buffer.position() == 0;
-            fileClassic.read(firstPageStartPosition, buffer);
+          try {
+            if (pageCount == 1) {
+              final ByteBuffer buffer = bufferPool.acquireDirect(false);
+              assert buffer.position() == 0;
 
-            if (verifyChecksums && (checksumMode == OChecksumMode.StoreAndVerify || checksumMode == OChecksumMode.StoreAndThrow))
-              verifyChecksum(buffer, fileId, startPageIndex, null);
+              fileClassic.read(firstPageStartPosition, buffer, false);
 
-            buffer.position(0);
+              if (verifyChecksums && (checksumMode == OChecksumMode.StoreAndVerify || checksumMode == OChecksumMode.StoreAndThrow))
+                verifyChecksum(buffer, fileId, startPageIndex, null);
```

Comment:
Actually, it's released in `verifyChecksum`, if `buffersToRelease` parameter value is `null` then buffer pointed by `buffer` parameter is released. It's an optimization, so we don't allocate additional array if we read only one page, which is the most common case.